notebook/notebookapp.py

```diff
@@ -250,9 +251,12 @@ def init_settings(self, jupyter_app, kernel_manager, contents_manager,
             # collapse $HOME to ~
             root_dir = '~' + root_dir[len(home):]
 
+        # Use the NotebookApp logger and its formatting for tornado request logging.
+        log_function = functools.partial(
+            log_request, log=log, log_json=jupyter_app.log_json)
```

Comment:
Are you sure this is necessary?  Prior to your very latest change, I had found the original 'partial' didn't change anything.  I'll need to verify this version of the change, but wanted to get my review comments submitted.

Reply:
Without this the `log_request` function would only get the `handler` argument because that's all tornado will pass to it. Nothing blows up but you don't get JSON formatting on what gets logged in `log_request`.

To compare, if I remove the partial and just pass `log_function=log_request` as before we get something like this:

```json
{"written_at": "2020-10-12T16:29:47.157Z", "written_ts": 1602520187157654000, "msg": "404 GET /nbextensions/widgets/notebook/js/extension.js (127.0.0.1) 2.770000ms referer=http://localhost:8888/notebooks/Untitled.ipynb", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "WARNING", "module": "log", "line_no": 52}
```

But with the change and with JSON logging enabled, we get something like this:

```json
{"written_at": "2020-10-12T16:31:16.698Z", "written_ts": 1602520276698143000, "msg": "", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "WARNING", "module": "log", "line_no": 50, "status": 404, "method": "GET", "ip": "127.0.0.1", "uri": "/nbextensions/widgets/notebook/js/extension.js", "request_time": 7.66, "referer": "http://localhost:8888/notebooks/Untitled.ipynb"}
```

In the former, note that the request fields are in the `msg` field, but in the latter the `ns` request fields are split out as specific fields in the log record which is the goal (for me anyway) so you can build queries in a distributed logging system like ELK or LogDNA and filter on things like `request_time:>=30`.

Reply:
I see - thank you.